.github/workflows/test-pypi.yml

```diff
@@ -0,0 +1,11 @@
+name: cd
+
+on:
+  workflow_dispatch: ~
```

Comment:
@tmi shouldn't we also trigger this on PR to main?

Reply:
this is the `test-pypi` action, I don't think we need regular triggers for it...

Reply:
I think it's a good test! I've seen the deployment to pypi fail many times. This will ensure everything is well packaged before creating the tag. And we only merge to main before a release anyway, which doesn't happen every day, that's why I was suggesting main, not develop.

Reply:
ah, you meant on _opening_ the PR, not pushing to main, I misread you. That does make sense